hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -340,8 +344,13 @@ public void sendRequest(byte[] buffer, int offset, int length) throws IOExceptio
            If expect header is not enabled, we throw back the exception.
          */
         String expectHeader = getConnProperty(EXPECT);
-        if (expectHeader != null && expectHeader.equals(HUNDRED_CONTINUE)) {
-          LOG.debug("Getting output stream failed with expect header enabled, returning back ", e);
+        if (expectHeader != null && expectHeader.equals(HUNDRED_CONTINUE)
+            && e instanceof ProtocolException
+            && EXPECT_100_JDK_ERROR.equals(e.getMessage())) {
+          LOG.debug(
+              "Getting output stream failed with expect header enabled, returning back ",
+              e);
```

Comment:
351 and 352 lines can be merged

Reply:
sure. It was done by IDE. I think keeping as before shouldnt be a problem. will take this.